src/framework/server.ts

```diff
@@ -76,8 +61,46 @@ export type SettingsInput = createExpressGraphql.OptionsData &
     plugins: Plugin.RuntimeContributions[]
     contextContributors: ContextContributor<any>[]
   }
+/**
+ * [API Reference](https://nexus-future.now.sh/#/references/api?id=server)  ⌁  [Guide](todo)
+ *
+ * ### todo
+ *
+ */
+export interface Server {
+  /**
+   * todo
+   */
+  start: () => Promise<void>
+  /**
+   * todo
+   */
+  stop: () => Promise<void>
+}
+
+interface ServerWithInstance<T extends any> extends Server {
+  instance: T
+}
+
+interface CustomServerHookInput {
+  schema: GraphQL.GraphQLSchema
+  defaultServer: ServerWithInstance<Express>
+  settings: App.SettingsData['server']
```

Comment:
Calling this `serverSettings` would be more consistent with other name of `defaultServer`?

Reply:
Renamed too! I agree with you, I was initially passing the whole `settings` object. If I may be really neat-picking here, I don't like the fact that we have `<server>Settings` and `default<Server>` (server is used as prefix and as suffix).
Makes me wonder if it'd be better to scope everything under a single `server` property.

```
server: {
  originalInstance: Express
  settings: SettingsData['server']
}
```

Reply:
Ah, just saw your comment below. Yeah, you're right, let's just go with `express` and let them take the settings the import